applicationset/controllers/applicationset_controller.go

```diff
@@ -357,6 +376,62 @@ func (r *ApplicationSetReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 	}, nil
 }
 
+func (r *ApplicationSetReconciler) performReverseDeletion(ctx context.Context, logCtx *log.Entry, appset argov1alpha1.ApplicationSet, currentApps []argov1alpha1.Application) (time.Duration, error) {
+	requeueTime := 10 * time.Second
+	// get deletionOrder using current Apps
+	stepLength := len(appset.Spec.Strategy.RollingSync.Steps)
+	if stepLength == 0 {
```

Comment:
I think this check should be first task in the function and there should be an early exit.
```
if len(appset.Spec.Strategy.RollingSync.Steps) <=0 {
logCtx.Info("no rolling sync steps found")
return
}
```